builtin/providers/azurerm/resource_arm_network_interface.go

```diff
@@ -0,0 +1,393 @@
+	}
+
+	resGrp := id.ResourceGroup
+	name := id.Path["networkInterfaces"]
+
+	// fetch the interface off Azure:
+	iface, err := ifaceClient.Get(resGrp, name)
+	if iface.StatusCode == http.StatusNotFound {
+		d.SetId("")
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("Error reading the state of network interface %q off Azure: %s", name, err)
+	}
+
+	// now update all the variable fields:
+	props := *iface.Properties
+
+	d.Set("vm_id", *props.VirtualMachine.ID)
+	d.Set("network_security_group_id", *props.NetworkSecurityGroup.ID)
+	d.Set("mac_address", *props.MacAddress)
+
+	// get the ip configs:
+	var ipConfigs []map[string]interface{}
+	for _, ipconf := range *props.IPConfigurations {
+		v := map[string]interface{}{}
+
+		v["id"] = *ipconf.ID
+		v["name"] = *ipconf.Name
+
+		// check for the allocation method and the address it could imply:
+		if ipconf.Properties.PrivateIPAllocationMethod == network.Static {
+			v["private_ip_address"] = *ipconf.Properties.PrivateIPAddress
+			v["dynamic_private_ip"] = false
+		} else { // guaranteed to be network.Dynamic
+			v["private_ip_address"] = ""
+			v["dynamic_private_ip"] = true
+		}
+
+		v["subnet_id"] = *ipconf.Properties.Subnet.ID
+		v["public_ip_id"] = *ipconf.Properties.PublicIPAddress.ID
+
+		// get the load balancer backpools and inbound nat rules:
+		var backpools []string
+		if bps := ipconf.Properties.LoadBalancerBackendAddressPools; bps != nil {
+			for _, bp := range *bps {
+				backpools = append(backpools, *bp.ID)
+			}
+
+			v["load_balancer_backend_pool_ids"] = backpools
+		} else {
+
+		}
+
+		var natrules []string
+		if nrs := ipconf.Properties.LoadBalancerInboundNatRules; nrs != nil {
+			for _, nr := range *nrs {
+				natrules = append(natrules, *nr.ID)
+			}
+
+			v["load_balancer_inbound_nat_rule_ids"] = natrules
+		}
+
+		// and finally, append the new list element:
+		ipConfigs = append(ipConfigs, v)
+	}
+	d.Set("ipConfig", ipConfigs)
+
+	// and finally; read the DNS settings:
+	d.Set("dns_servers", *iface.Properties.DNSSettings.DNSServers)
+	d.Set("applied_dns_servers", *iface.Properties.DNSSettings.AppliedDNSServers)
+	d.Set("internal_name", *iface.Properties.DNSSettings.InternalDNSNameLabel)
+	d.Set("internal_fqdn", *iface.Properties.DNSSettings.InternalFqdn)
+
+	return nil
+}
+
+// resourceArmNetworkInterfaceDelete deletes the specified ARM network interface.
+func resourceArmNetworkInterfaceDelete(d *schema.ResourceData, meta interface{}) error {
+	ifaceClient := meta.(*ArmClient).ifaceClient
+
+	// fetch the name of the network interface and the containing
+	// resource group from the given parser.
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return fmt.Errorf("Error parsing ID of ARM network interface: %s")
+	}
+	resGrp := id.ResourceGroup
+	name := id.Path["networkInterfaces"]
+
+	// then, issue the actual deletion:
+	resp, err := ifaceClient.Delete(resGrp, name)
+	if resp.StatusCode == http.StatusNotFound {
```

Comment:
@phinze: wanted confirmation that if the instance got deleted already when we issue one if it's an error or a lucky break?